dataset/include/scipp/dataset/dataset.h

```diff
@@ -32,6 +33,18 @@ struct DatasetData {
   /// Unaligned coords.
   std::unordered_map<Dim, Variable> coords;
   std::unordered_map<std::string, Variable> masks;
+
+  scipp::index sizeInMemory() const {
```

Comment:
Have you considered computing size based on `data()`, `aligned_coords()`, `unaligned_coords()`, and `masks()`? This would avoid messing to much with implementation details.